include/net/socket.h

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __NET_SOCKET_H
+#define __NET_SOCKET_H
+
+#include <sys/types.h>
+#include <zephyr/types.h>
+#include <net/net_ip.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+int zsock_socket(int family, int type, int proto);
+int zsock_close(int sock);
+int zsock_bind(int sock, const struct sockaddr *addr, socklen_t addrlen);
+int zsock_connect(int sock, const struct sockaddr *addr, socklen_t addrlen);
+int zsock_listen(int sock, int backlog);
+int zsock_accept(int sock, struct sockaddr *addr, socklen_t *addrlen);
+ssize_t zsock_send(int sock, const void *buf, size_t len, int flags);
+ssize_t zsock_recv(int sock, void *buf, size_t max_len, int flags);
```

Comment:
> Independent of where socket offloading occurs, it just seems it would be logical and future-proof to have zsocket.h defining native zsock functions, and socket.h declaring the socket functions

Gil, I appreciate the idea, but after additional checking, I don't see such pattern of splitting the same API across 2+ headers to be used in Zephyr. And as I mentioned above, these are indeed intended to be 2 facets of the same API, not 2 different APIs. I'll be happy to make such a change however is there will be 2nd vote for it. Otherwise, well, if it proves to be useful/need, we may need to do such a refactor in the future (e.g. when we'll have better defined POSIX API layer). Thanks for understanding.